internal/server/v2/parser.go

```diff
@@ -118,8 +116,52 @@ func ParseArc(s string) (*Arc, error) {
 		return arc, nil
 	}
 	// No '+' or '{' found, this is a single property.
-	if len(s) > 0 {
-		arc.SingleProp = s
+	if len(expr) > 0 {
+		arc.SingleProp = expr
 	}
 	return arc, nil
 }
+
+// ParseProperty parses an expression string into a list of Arcs.
+func ParseProperty(expr string) ([]*Arc, error) {
+	parts := splitExpr(expr)
+	if len(parts) == 1 {
+		// Handle "->" query, which is to get all properties
+		parts = append(parts, "")
+	}
+	if len(parts)%2 == 1 {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "in valid expression string: %s", expr)
+	}
+	arcs := []*Arc{}
+	for i := 0; i < len(parts)/2; i++ {
+		arc, err := parseArc(parts[i*2], parts[i*2+1])
+		if err != nil {
+			return nil, err
+		}
+		arcs = append(arcs, arc)
+	}
+	return arcs, nil
+}
+
+// ParseGraph parses an expression string into a Graph object
+func ParseGraph(expr string) (*Graph, error) {
+	parts := splitExpr(expr)
+	if len(parts) < 3 || len(parts)%2 == 0 {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "in valid expression string: %s", expr)
+	}
+	g := &Graph{
+		Subject: parts[0],
+	}
+	arcs := []*Arc{}
+	for i := 0; i < len(parts)/2; i++ {
+		arc, err := parseArc(parts[i*2+1], parts[i*2+2])
+		if err != nil {
+			return nil, err
+		}
+		arcs = append(arcs, arc)
+	}
+	g.Arcs = arcs
```

Comment:
Don't need to define 'arcs' here, just re-use g.Arcs:

	for i := 0; i < len(parts)/2; i++ {
		arc, err := parseArc(parts[i*2+1], parts[i*2+2])
		if err != nil {
			return nil, err
		}
		g.Arcs = append(arcs, arc)
	}